src/str/dissimilar.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2021 The predicates-rs Project Developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::borrow;
+use std::fmt;
+
+use crate::reflection;
+use crate::Predicate;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+enum DistanceOp {
+    Similar,
+    Different,
+}
+
+impl DistanceOp {
+    fn eval(self, limit: i32, distance: i32) -> bool {
+        match self {
+            DistanceOp::Similar => distance <= limit,
+            DistanceOp::Different => limit < distance,
+        }
+    }
+}
+
+/// Predicate that diffs two strings.
+///
+/// This is created by [predicate::str::similar2] or [predicate::str::diff2].
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DissimilarPredicate {
+    orig: borrow::Cow<'static, str>,
+    distance: i32,
+    op: DistanceOp,
+}
+
+impl DissimilarPredicate {
+    /// The split used when identifying changes.
+    ///
+    /// # Deprecated
+    ///
+    /// No longer has any effect. The diff algorithm should find the best split automatically.
+    #[deprecated(since="1.1.0" note="No longer needed, diff algorithm should find the best split")]
+    pub fn split<S>(mut self, split: S) -> Self
+    where
+        S: Into<borrow::Cow<'static, str>>,
+    {
+        self
+    }
+
+    /// The maximum allowed edit distance.
+    ///
+    /// Default: `0`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use predicates::prelude::*;
+    ///
+    /// let predicate_fn = predicate::str::similar2("Hello World!").distance(1);
+    /// assert_eq!(true, predicate_fn.eval("Hello World!"));
+    /// assert_eq!(true, predicate_fn.eval("Hello World"));
+    /// assert_eq!(false, predicate_fn.eval("Hello World?"));
+    /// ```
+    pub fn distance(mut self, distance: i32) -> Self {
+        self.distance = distance;
+        self
+    }
+
+    fn diff(&self, chunks: &Vec<dissimilar::Chunk<'_>>) -> String {
+        use std::fmt::Write;
+        let mut f = String::with_capacity(chunks.len());
+        for c in chunks {
+            match *c {
+                dissimilar::Chunk::Equal(ref s) => write!(f, "{}", s),
+                dissimilar::Chunk::Delete(ref s) => write!(f, "\x1b[92m{}\x1b[0m", s),
+                dissimilar::Chunk::Insert(ref s) => write!(f, "\x1b[91m{}\x1b[0m", s),
```

Comment:
Unsure if it needs to be in this PR or not but I'd like to switch to a crate for handling color